app/models/data_resources/resource_modules/public_transportation/travel_time.rb

```diff
@@ -0,0 +1,128 @@
+
+    # get weekday of date string "2023-09-18T12:00"
+    @weekday_name = @travel_date.strftime("%A").downcase
+    @calendar_exception_date = @travel_date.strftime("%Y%m%d")
+    @travel_time_from = @travel_date.strftime("%H:%M:%S")
+
+    @calculated_travel_times = []
+  end
+
+  def travel_times
+    return [] if @external_id.blank?
+
+    stop_times = RedisAdapter.get_gtfs_stop_times(@external_id, @data_provider_id)
+    stop_times.each do |stop_time|
+      stop_time_data = gtfs_stop_time_data(stop_time)
+      trip = RedisAdapter.get_gtfs_trip(stop_time["trip_id"], @data_provider_id)
+      if trip.present?
+        stop_time_data["trip"] = gtfs_trip_data(trip)
+        stop_time_data["route"] = gtfs_route(trip["route_id"])
+
+        traveling = gtfs_calendar(stop_time, trip["service_id"])
+        @calculated_travel_times << stop_time_data if traveling
+      end
+    end
+
+    @calculated_travel_times
+  end
+
+  private
+
+  def gtfs_calendar(stop_time, service_id)
+    gtfs_cal_data = RedisAdapter.get_gtfs_calendar(service_id, @data_provider_id)
+    traveling = gtfs_cal_data[@weekday_name].to_i
+    exception = RedisAdapter.get_gtfs_calendar_dates(service_id, @calendar_exception_date, @data_provider_id) || nil
+
+    case exception.to_i
+    when 1
+      traveling = 1
+    when 2
+      traveling = 0
+    end
+
+    traveling = 0 if @calendar_exception_date.to_i < gtfs_cal_data["start_date"].to_i
+    traveling = 0 if @calendar_exception_date.to_i > gtfs_cal_data["end_date"].to_i
+
+    # check if travel time is after departure time
+    # departure_time kann auch "24:14" sein
+    begin
+      departure_time = Time.parse(stop_time["departure_time"])
+      travel_time_from = Time.parse(@travel_time_from)
+      traveling = 0 if departure_time.to_i < travel_time_from.to_i
+    rescue
+      # do nothing
+    end
+
+    traveling == 1
+  end
+
+  def gtfs_route(route_id)
+    route_data = RedisAdapter.get_gtfs_route(route_id, @data_provider_id)
+
+    {
+      route_short_name: route_data["route_short_name"],
+      route_long_name: route_data["route_long_name"],
+      route_type: route_data["route_type"],
+      route_color: route_data["route_color"],
+      route_text_color: route_data["route_text_color"],
+      route_desc: route_data["route_desc"]
+    }
+  end
+
+  def gtfs_trip_data(trip)
+    {
+      trip_headsign: trip["trip_headsign"],
+      trip_short_name: trip["trip_short_name"],
+      direction_id: trip["direction_id"],
+      wheelchair_accessible: trip["wheelchair_accessible"],
+      bikes_allowed: trip["bikes_allowed"]
+    }
+  end
+
+  def gtfs_stop_time_data(stop_time)
+    {
+      stop_sequence: stop_time["stop_sequence"],
+      stop_headsign: stop_time["stop_headsign"],
+      pickup_type: stop_time["pickup_type"],
+      drop_off_type: stop_time["drop_off_type"],
+      arrival_time: stop_time["arrival_time"],
+      departure_time: stop_time["departure_time"]
+    }
+  end
+end
```

Comment:
Abfahrtszeiten werden aus der Redis DB ausgelesen und in ein passendes JSON Format gebracht, da nicht alle GTFS Daten werden derzeit in der App benötigt werden.